### PR TITLE
Register authentication 2

### PR DESCRIPTION
**Pull request by sepehr9877**
@SoroorAghamiri  the user is saved on db after registration and will be redirect to the profile page, the authentication is working for both login and registration however i couldnt handle duplicate username error yet.

### CODE REVIEW

**Comment by SoroorAghamiri** [comment]
Your previous pull request works fine. I'm closing this one.

**Review by SoroorAghamiri — APPROVED** [review]

File: src/main/java/ca/concordia/soen6011/configuration/MyAuthenticationProvider.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.web.authentication.AuthenticationSuccessHandler;
+import org.springframework.stereotype.Component;
+
+@Component
+public class MyAuthenticationProvider implements AuthenticationProvider {
+
+	@Autowired
+    private UserDetailsService userDetailsService;
```
Comment (MajorTreble) [inline src/main/java/ca/concordia/soen6011/configuration/MyAuthenticationProvider.java:19]: spacing is off

File: src/main/java/ca/concordia/soen6011/configuration/MyAuthenticationProvider.java [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+	@Override
+	public Authentication authenticate(Authentication authentication) throws AuthenticationException {
+		String username = authentication.getName();
+        String password = authentication.getCredentials().toString();
```
Comment (MajorTreble) [inline src/main/java/ca/concordia/soen6011/configuration/MyAuthenticationProvider.java:23]: indent

File: src/main/java/ca/concordia/soen6011/configuration/WebSecurityConfig.java [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+import org.springframework.security.config.annotation.web.configuration.WebSecurityConfiguration;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.web.SecurityFilterChain;
+import org.springframework.security.web.authentication.AuthenticationSuccessHandler;
+
+@EnableWebSecurity
+@Configuration
+public class WebSecurityConfig  {
+//
```
Comment (MajorTreble) [inline src/main/java/ca/concordia/soen6011/configuration/WebSecurityConfig.java:20]: remove comments

**Review by MajorTreble — APPROVED** [review]

**Pull request merged**